cli/auth.go

```diff
@@ -320,6 +320,180 @@ func (c *viamClient) createOrganizationAPIKey(orgID, keyName string) (*apppb.Cre
 	return c.client.CreateKey(c.c.Context, req)
 }
 
+// LocationAPIKeyCreateAction corresponds to `location api-key create`.
+func LocationAPIKeyCreateAction(cCtx *cli.Context) error {
+	c, err := newViamClient(cCtx)
+	if err != nil {
+		return err
+	}
+
+	err = c.locationAPIKeyCreateAction(cCtx)
+	return err
+}
+
+func (c *viamClient) locationAPIKeyCreateAction(cCtx *cli.Context) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	locationID := cCtx.String(dataFlagLocationID)
+	orgID := cCtx.String(dataFlagOrgID)
+	keyName := cCtx.String(apiKeyCreateFlagName)
+
+	if locationID == "" {
+		return errors.New("cannot create an api-key for a location without an ID")
+	}
+
+	if keyName == "" {
+		// Default name is in the form myusername@gmail.com-2009-11-10T23:00:00Z
+		// or key-uuid-2009-11-10T23:00:00Z if it was created by a key
+		keyName = fmt.Sprintf("%s-%s", c.conf.Auth, time.Now().Format(time.RFC3339))
+		infof(cCtx.App.Writer, "using default key name of %s", keyName)
+	}
+
+	if orgID == "" {
+		// try and create a key for the location based on the only org on the location
+		// if the user doesn't have perms for this org this won't work
+
+		loc, err := c.client.GetLocation(c.c.Context, &apppb.GetLocationRequest{
+			LocationId: locationID,
+		})
+		if err != nil {
+			return err
+		}
+
+		if loc.GetLocation() == nil {
+			return errors.Errorf("cannot create api-key for locationID: %s, empty location", locationID)
+		}
+
+		if len(loc.GetLocation().GetOrganizations()) != 1 {
+			return errors.Errorf("cannot create api-key for locationID: %s as there are mutiple orgs attached", locationID)
```

Comment:
Could elaborate here to mention using `dataFlagOrgID`.

Reply:
removed anyways